src/pci_lmt/config.py

```diff
@@ -66,8 +47,7 @@ def from_json(data: ty.Dict[str, ty.Any]) -> "ConfigLmtGroup":
             name=data["name"],
             receiver_number=data["receiver_number"],
             bdf_list=data["bdf_list"],
-            margin_type=data["margin_type"],
-            margin_direction=data["margin_direction"],
+            margin_type=MarginType(f"{data['margin_type'].lower()}_{data['margin_direction'].lower()}"),
```

Comment:
you may still want to validate inputs here

Reply:
Sure. Just to confirm my understanding:
The interpreter would throw an exception if the inputs are invalid. But we can add more context (in the error information) if we validate the inputs explicitly. Is that right?

Reply:
it's better to have explicit validation for inputs, than assume some operations will fail given invalid input later (which can result in weird errors, or can result in _no errors_ in some cases, both of which are undesirable). so yes, i am confirming your understanding